package/gluon-core/luasrc/lib/gluon/upgrade/140-firewall-rules

```diff
@@ -1,25 +1,87 @@
 #!/usr/bin/lua
+-- TODO: replace REJECT by DROP. only use REJECT for debugging
 
 local site = require 'gluon.site_config'
 local uci = require('simple-uci').cursor()
 
+uci:section('firewall', 'zone', 'wan', {
+	name = 'wan',
+	network = { 'wan','wan6',},
+	input = 'REJECT',
+	output = 'ACCEPT',
+	forward = 'REJECT',
+	mtu_fix = '1',
+	conntrack = '1',
+	log = '1',
+}
+)
+-- the client zone is set up by gluon-client-bridge
+--
+uci:section('firewall', 'zone', 'mesh',
+{
+	name = 'mesh',
+	input = 'REJECT',
+	output = 'ACCEPT',
+	forward = 'REJECT',
+	conntrack = '1',
+	log = '1',
+}
+)
 
-local function reject_input_on_wan(zone)
-	if zone.name == 'wan' then
-		uci:set('firewall', zone['.name'], 'input', 'REJECT')
-		uci:set('firewall', zone['.name'], 'conntrack', true)
-	end
+-- allow inbound ssh from anywhere
+for _, zone in ipairs({ 'wan', 'client', 'mesh' })
+do
+	uci:section('firewall', 'rule', zone .. '_ssh',
+	{
+		name =  zone .. '_ssh',
+		src = zone,
+		dest_port = '22',
+		proto = 'tcp',
+		target = 'ACCEPT',
+	}
+	)
+end
+
+
+-- allow icmp in/out/forward on all relevant zones
+for _, zone in ipairs ({ 'mesh', 'wan', 'client' } )
+do
+	uci:section('firewall','rule',  zone .. 'ICMPv6in',
+	{
+		name = 'Allow-ICMPv6-Input',
+		src = zone,
+		proto = 'icmp',
+		icmp_type = {'echo-request', 'echo-reply', 'destination-unreachable', 'packet-too-big', 'time-exceeded', 'bad-header', 'unknown-header-type', 'router-solicitation', 'neighbour-solicitation', 'router-advertisement', 'neighbour-advertisement', },
+		limit = '1000/sec',
+		family = 'ipv6',
+		target = 'ACCEPT',
+	}
+	)
+
+	uci:section('firewall','rule', zone .. 'ICMPv6out',
+	{
+		name = 'Allow-ICMPv6-OUTPUT',
+		dest = 'wan',
+		proto = 'icmp',
+		icmp_type = {'echo-request', 'echo-reply', 'destination-unreachable', 'packet-too-big', 'time-exceeded', 'bad-header', 'unknown-header-type', 'router-solicitation', 'neighbour-solicitation', 'router-advertisement', 'neighbour-advertisement' },
+		limit = '1000/sec',
+		family = 'ipv6',
+		target = 'ACCEPT',
+	}
+	)
 
-	return true
+	uci:section('firewall','rule', zone ..'ICMPv6forward',
+	{
+		name = 'Allow-ICMPv6-Forward',
+		src = zone,
+		dest = '*',
```

Comment:
No. Absolutely no traffic may ever flow between mesh and WAN. Generally, we don't need a special ICMP forward rule: for any pair of zones, we either forward all traffic or no traffic. ICMP traffic is not special with regard to forwarding.